demo-simple/src/main/java/com/novoda/downloadmanager/demo/MainActivity.java

```diff
@@ -116,144 +106,94 @@ public void onClick(View v) {
+        LocalFilesDirectory localFilesDirectory = LocalFilesDirectoryFactory.create(getApplicationContext());
+        for (String fileName : localFilesDirectory.contents()) {
+            Log.d("LogFileDirectory", fileName);
         }
     };
 
-    private final View.OnClickListener logFileDirectoryOnClick = new View.OnClickListener() {
-        @Override
-        public void onClick(View v) {
-            LocalFilesDirectory localFilesDirectory = LocalFilesDirectoryFactory.create(getApplicationContext());
-            for (String fileName : localFilesDirectory.contents()) {
-                Log.d("LogFileDirectory", fileName);
-            }
+    private final View.OnClickListener logLocalFilePathOfDownloadOnClick = v -> liteDownloadManagerCommands.getFirstLocalPathForDownloadWithMatching(
+            TWENTY_FILE_URL,
+            filePath -> Log.d("LocalUriForTwentyMBFile: ", filePath.path())
+    );
+
+    private final DownloadBatchCallback callback = downloadBatchStatus -> {
+        String status = getStatusMessage(downloadBatchStatus);
+
+        String message = "Batch " + downloadBatchStatus.getDownloadBatchTitle().asString()
+                + "\ndownloaded: " + downloadBatchStatus.percentageDownloaded() + "%"
+                + "\nbytes: " + downloadBatchStatus.bytesDownloaded()
+                + status
+                + "\n";
+
+        DownloadBatchId downloadBatchId = downloadBatchStatus.getDownloadBatchId();
+        if (downloadBatchId.equals(BATCH_ID_1)) {
+            textViewBatch1.setText(message);
+        } else if (downloadBatchId.equals(BATCH_ID_2)) {
+            textViewBatch2.setText(message);
         }
     };
 
-    View.OnClickListener logLocalFilePathOfDownloadOnClick = new View.OnClickListener() {
-        @Override
-        public void onClick(View v) {
-            liteDownloadManagerCommands.getFirstLocalPathForDownloadWithMatching(TWENTY_FILE_URL, new DownloadFilePathCallback() {
-                @Override
-                public void onReceived(FilePath filePath) {
-                    Log.d("LocalUriForTwentyMBFile: ", filePath.path());
-                }
-            });
+    private String getStatusMessage(DownloadBatchStatus downloadBatchStatus) {
+        if (downloadBatchStatus.status() == ERROR) {
+            return "\nstatus: " + downloadBatchStatus.status()
+                    + " - " + downloadBatchStatus.getDownloadErrorType();
+        } else {
+            return "\nstatus: " + downloadBatchStatus.status();
         }
-    };
+    }
 
-    private final AllBatchStatusesCallback batchStatusesCallback = new AllBatchStatusesCallback() {
-        @Override
-        public void onReceived(List<DownloadBatchStatus> downloadBatchStatuses) {
-            for (DownloadBatchStatus downloadBatchStatus : downloadBatchStatuses) {
-                callback.onUpdate(downloadBatchStatus);
-            }
+    private final AllBatchStatusesCallback batchStatusesCallback = downloadBatchStatuses -> {
```

Comment:
I think it'd be nicer if this was a method and then use a method reference instead

Reply:
there's many places that are using fields instead of methods, so I'm fine either way if you don't want to make that many change